dashboard/urls.py

```diff
@@ -39,26 +39,26 @@
     # front-end will manage any additional routing.
     re_path(r'^courses/', login_required(views.get_home_template,), name="courses"),
     # This path is used by Course.get_absolute_url() to generate course links for the Admin interface.
-    path('courses/<int:course_id>', login_required(views.get_home_template,), name="courses"),
+    path('courses/<int:canvas_id>', login_required(views.get_home_template,), name="courses"),
```

Comment:
I can't speak to the other `urlpattern` changes, but this is breaking `Course.get_absolute_url()` (and the Django admin interface). I think you need to change the keyword in `models.py` to be `canvas_id`.

Reply:
Yeah, I was afraid of this change. You can't later change the type of a variable you've defined so I had to change this to a different variable. I kind of was thinking of just doing all of these on a separate future PR as they had a chance of breaking once I started touching the hints there.